### PR TITLE
Remove doctests and add support for Numeric/String FeatureID

### PR DESCRIPTION
**Pull request by Ozmat**
- Add support for numeric and string feature ids
- Turn all doctests into Tasty tests
- Expose all modules and specify the Internal ones

### CODE REVIEW

**Comment by domdere** [comment]
Cool, thanks, I'll review ASAP

File: test/Data/Geospatial/Internal/CRSTests.hs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.Geospatial.Internal.CRSTests where
```
Comment (domdere) [inline test/Data/Geospatial/Internal/CRSTests.hs]: Thanks for removing doctest!

It would have been quite a time consuming exercise

**Comment by Ozmat** [comment]
Thanks !

Yes I was going to suggest you to update your CI file and use stack to build/test, something like :

```yaml
pre:
  - wget https://github.com/commercialhaskell/stack/releases/download/v1.6.1/stack-1.6.1-linux-x86_64.tar.gz -O /tmp/stack.tar.gz
  - mkdir /tmp/stack/
  - tar -xvzf /tmp/stack.tar.gz -C /tmp/stack/
  - export STACK=/tmp/stack/stack-1.6.1-linux-x86_64/stack

setup:
  - $STACK setup --no-terminal
  # for dependencies, will fail faster if there is a problem
  - $STACK build --no-terminal --only-configure

compile:
  - $STACK build --no-terminal

test:
  - $STACK test --no-terminal
```

If you want to have faster builds (I saw that you were checking for a system GHC in your current CI file) you can try this one :

```
$STACK build --system-ghc
```

which will actually check if the system version is compatible with the project configuration (particularly the resolver setting).

Otherwise you can use [cache directories](https://docs.travis-ci.com/user/caching/) and only download GHC once (or each time the version/resolver changes).

```yaml
- cache_directories:
  - "~/.stack"
```
You can probably cache your "./.stack-work" and avoid rebuilding dependencies if you really care about the build time (not sure about this one though).

**Comment by domdere** [comment]
hmm ok I was hoping to keep the `stack` dependence to a minimum unfortunately.

I actually regret adding a `stack.yaml` file to the project at all, I can fix up the CI in a seperate pass myself.

It could do with an update to make it more efficient, so I may as well do that at the same time.

**Comment by Ozmat** [comment]
No It wasn't a mistake. We were actually using the original one (or maybe a fork that was using a newer lts), and needed support for the numeric FeatureID so we just implemented it a month ago, and I spent  some time yesterday converting the test because It wasn't working with doctests (and doctests-discover).

**Comment by domdere** [comment]
I'll probably just update it to use my [current standard CI yaml](https://github.com/domdere/ultra/blob/master/.travis.yml)

**Comment by domdere** [comment]
ok, thanks @Ozmat I'll merge this in and do a pass to update the CI build

**Comment by domdere** [comment]
It might take a while for this to hit Hackage though

**Comment by domdere** [comment]
(Apologies btw, this project doesnt get much love from me)

**Comment by domdere** [comment]
@Ozmat If sitewisely want to take it over btw let me know.

Otherwise, the  "half in, half out" approach with `stack` that I would take will look a little inconsistent, thats how much I despise `stack` unfortunately (I dont use it in my other projects at all).

**Comment by Ozmat** [comment]
Sounds good, thanks ! No worries, we're using our fork for now and added some extra love for you ;)

Yep sure ! You can just give the collaborator access to @newmana and @Ozmat, it should be alright.

**Comment by domdere** [comment]
Ok added you two as collaborators, I'm also fine if sitewisely wants to get rid of their fork and I can transfer ownership of the repo to that organisation, so you aren't beholden to me in any way

(can also sort out the hackage maintainer privs)

**Comment by newmana** [comment]
Wasn't sure the best way to contact you - but we've finally gotten around to getting rid of our fork and we'd be happy to become the hackage maintainer.

**Pull request merged**